server/messages.go

```diff
@@ -182,20 +190,11 @@ func sendToPacketInfoChan(b *bytes.Buffer, conn net.Conn, c chan *packetInfo, t
 
 	err := proto.Unmarshal(b.Bytes(), pdata)
 	if err != nil {
-		if debugMode {
-			fmt.Println("[Error] Unmarshal failed:", b.Bytes())
-		}
+		log.Debugf(logCommunication+"Unmarshal failed: %v\n", b.Bytes())
 		return err
 	}
 
-	if debugMode {
-		fmt.Println("[Received]", pdata)
-		jsonData, err := json.MarshalIndent(pdata, "", "  ")
-		if err != nil {
-			return err
-		}
-		fmt.Printf("%s\n", jsonData)
-	}
+	log.Debugf("[Received] from %s: %s\n", getIP(conn), pdata)
```

Comment:
How about `log.Debugf(logCommunication+"Received data from %s: %s\n", getIP(conn), pdata)`

Reply:
Yes. I should have fixed that when I did the other bare string singleton.

Reply:
24d569e